swim.go

```diff
@@ -279,33 +308,201 @@ func (s *SWIM) startFailureDetector() {
 // 3. At the end of T, SWIM checks to see if ack was received from k members, and if there is no message,
 //    The member(j) is judged to be failed, so check the member(j) as suspected or delete the member(j) from memberMap.
 //
-
-func (s *SWIM) probe(member Member) {
+func (s *SWIM) probe(member Member, timer *time.Timer) {
 
 	if member.Status == Dead {
 		return
 	}
 
-	end := make(chan struct{}, 1)
-	defer close(end)
+	end := make(chan ProbeResponse, 1)
+	defer func() {
+		close(end)
+	}()
 
 	go func() {
+		err := s.ping(&member)
+		if err == nil {
+			end <- ProbeResponse{}
+			return
+		}
+		if err != ErrSendTimeout {
+			iLogger.Errorf(nil, "Error occurred when ping failed [%s]", err.Error())
```

Comment:
Do not handle the sendtimeout error?

Reply:
yes in the case of receiving timeout error we proceed to indirect-probing procedure